runtime/src/hmac.rs

```diff
@@ -81,20 +81,28 @@ fn ecc384_key_gen(
 pub enum Hmac {}
 
 impl Hmac {
-    /// "Hash" the data in the provided KV slot by HMACing it with an empty slice.
-    /// This mechanism is necessary because the hardware does not directly support
-    /// hashing data in KV slots.
+    /// Calculate HMAC-384 KDF
     ///
     /// # Arguments
     ///
     /// * `drivers` - Drivers
-    /// * `input` - KeyId containing the input data
-    /// * `output` - KeyId which the output hash should be written to
+    /// * `key` - HMAC384 key slot
+    /// * `label` - Input label
+    /// * `context` - Input context
+    /// * `output` - Key slot to store the output
     #[cfg_attr(not(feature = "no-cfi"), cfi_impl_fn)]
-    pub fn hmac384_hash(drivers: &mut Drivers, input: KeyId, output: KeyId) -> CaliptraResult<()> {
-        drivers.hmac384.hmac(
-            &KeyReadArgs::new(input).into(),
-            &HmacData::Slice(&[]),
+    pub fn hmac384_kdf(
```

Comment:
Is this code even used, or plan on being used?

With #1819, will this need to be updated to HMAC-SHA-512?

Reply:
This function was added a bit ago in anticipation of using it to use/test the hash chain feature. However the hash chain implementation never landed. Once the outstanding PRs are merged I can send the implementation for review, where this will be used.

And yeah we may as well make the hash chain use SHA512.